src/libraries/System.IO.Packaging/tests/Tests.cs

```diff
@@ -3961,6 +3961,23 @@ public void Roundtrip_Compression_Option(CompressionOption createdCompressionOpt
             }
         }
 
+        [Fact]
+        public void Cannot_Modify_Package_On_Unseekable_Stream()
+        {
+            var ba = File.ReadAllBytes("plain.docx");
+
+            using (MemoryStream ms = new MemoryStream())
+            {
+                ms.Write(ba, 0, ba.Length);
+                ms.Seek(0, SeekOrigin.Begin);
+
+                using (WrappedStream ws = new WrappedStream(ms, true, true, false))
+                {
+                    Assert.Throws<ArgumentException>(() => Package.Open(ws, FileMode.Open, FileAccess.ReadWrite));
```

Comment:
Looks good.

The Package.Open method is [here](https://source.dot.net/#System.IO.Packaging/System/IO/Packaging/Package.cs,886), and it then sets the `ZipArchiveMode` to `Update` [here](https://source.dot.net/#System.IO.Packaging/System/IO/Packaging/ZipPackage.cs,326), which is expected to throw [here](https://source.dot.net/#System.IO.Compression/System/IO/Compression/ZipArchive.cs,151).